amundsen_application/static/js/ducks/bookmark/reducer.ts

```diff
@@ -1,81 +1,71 @@
+import { Bookmark } from 'interfaces';
+
 import {
   AddBookmark,
   AddBookmarkRequest,
-  AddBookmarkResponse,
-  Bookmark,
   GetBookmarks,
-  GetBookmarksForUser,
-  GetBookmarksForUserRequest,
-  GetBookmarksForUserResponse,
   GetBookmarksRequest,
   GetBookmarksResponse,
+  GetBookmarksForUser,
+  GetBookmarksForUserRequest,
   RemoveBookmark,
   RemoveBookmarkRequest,
   RemoveBookmarkResponse,
-} from "./types";
-
-export type BookmarkReducerAction =
-  AddBookmarkRequest | AddBookmarkResponse |
-  GetBookmarksRequest | GetBookmarksResponse |
-  GetBookmarksForUserRequest | GetBookmarksForUserResponse |
-  RemoveBookmarkRequest | RemoveBookmarkResponse;
-
+} from './types';
 
+/* ACTIONS */
 export function addBookmark(resourceKey: string, resourceType: string): AddBookmarkRequest {
   return {
     resourceKey,
     resourceType,
-    type: AddBookmark.ACTION,
+    type: AddBookmark.REQUEST,
   }
-}
-
+};
 export function removeBookmark(resourceKey: string, resourceType: string): RemoveBookmarkRequest {
   return {
     resourceKey,
     resourceType,
-    type: RemoveBookmark.ACTION,
+    type: RemoveBookmark.REQUEST,
   }
-}
-
+};
 export function getBookmarks(): GetBookmarksRequest {
   return {
-    type: GetBookmarks.ACTION
+    type: GetBookmarks.REQUEST,
   }
-}
-
+};
 export function getBookmarksForUser(userId: string): GetBookmarksForUserRequest {
   return {
     userId,
-    type: GetBookmarksForUser.ACTION,
+    type: GetBookmarksForUser.REQUEST,
   }
-}
+};
 
- export interface BookmarkReducerState {
+/* REDUCER */
+export interface BookmarkReducerState {
   myBookmarks: Bookmark[];
   myBookmarksIsLoaded: boolean;
   bookmarksForUser: Bookmark[];
 }
 
- const initialState: BookmarkReducerState = {
+export const initialState: BookmarkReducerState = {
   myBookmarks: [],
   myBookmarksIsLoaded: false,
   bookmarksForUser: [],
 };
 
- export default function reducer(state: BookmarkReducerState = initialState, action: BookmarkReducerAction): BookmarkReducerState {
+ export default function reducer(state: BookmarkReducerState = initialState, action): BookmarkReducerState {
   switch(action.type) {
     case RemoveBookmark.SUCCESS:
-      const { resourceKey } = action.payload;
+      const { resourceKey } = (<RemoveBookmarkResponse>action).payload;
```

Comment:
nit: There's probably a way to let TS infer `action` type without casting. Not sure what's causing this to fail here.

Reply:
The type checker doesn't fail without this specification. Rather, our code isn't type safe without this specification. Without it I could write something like `const { resourceKey } = action.payload.invalidAttribute` and the type checker would not catch this mistake.

I believe there are other approaches, I just haven't figured out the best way to leverage the TypeScripts ability to infer types when using `switch/case` or if the support is fully there yet. Another hack could be to use `if` statement here, but I opted for the casting. When I'm done with this work, I can open issue for a better/cleaner approach to type safe actions.